src/components/screens/vacation-requests/vacation-requests-table/vacation-requests-table-toolbar/vacation-requests-table-toolbar.tsx

```diff
@@ -0,0 +1,188 @@
+import { GridRowId } from "@mui/x-data-grid";
+import { DataGridRow, ToolbarFormModes, VacationData } from "../../../../../types";
+import ToolbarDeleteButton from "./toolbar-delete-button";
+import FormToggleButton from "./toolbar-form-toggle-button";
+import ConfirmationHandler from "../../../../contexts/confirmation-handler";
+import strings from "../../../../../localization/strings";
+import { getToolbarTitle } from "../../../../../utils/toolbar-utils";
+import { VacationRequestStatuses } from "../../../../../generated/client";
+import UpdateStatusButton from "./toolbar-update-status-button";
+
+/**
+ * Component properties
+ */
+interface Props {
+  deleteVacationRequests: (selectedRowIds: GridRowId[], rows: DataGridRow[]) => Promise<void>;
+  createVacationRequest: (vacationData: VacationData) => Promise<void>;
+  updateVacationRequest: (
+    vacationData: VacationData,
+    vacationRequestId: string | undefined
+  ) => Promise<void>;
+  updateVacationRequestStatuses: (
+    newStatus: VacationRequestStatuses,
+    selectedRowIds: GridRowId[]
+  ) => Promise<void>;
+  setFormOpen: (formOpen: boolean) => void;
+  formOpen: boolean;
+  selectedRowIds: GridRowId[];
+  rows: DataGridRow[];
+}
+/**
+ * Table toolbar component
+ *
+ * @param props component properties
+ */
+const TableToolbar = ({
+  deleteVacationRequests,
+  createVacationRequest,
+  updateVacationRequest,
+  updateVacationRequestStatuses,
+  setFormOpen,
+  formOpen,
+  selectedRowIds,
+  rows
+}: Props) => {
+  const [toolbarOpen, setToolbarOpen] = useState<boolean>(false);
+  const [toolbarFormMode, setToolbarFormMode] = useState<ToolbarFormModes>(ToolbarFormModes.NONE);
+  const [confirmation, setConfirmation] = useState("");
+
+  /**
+   * Set toolbar open
+   */
+  useEffect(() => {
+    if (selectedRowIds) {
+      setToolbarOpen(true);
+    } else {
+      setToolbarOpen(false);
+    }
+  }, [selectedRowIds]);
+
+  /**
+   * Delete vacation requests and statuses
+   */
+  const deleteVacationsData = async () => {
+    await deleteVacationRequests(selectedRowIds, rows);
+  };
+
+  /**
+   * Toolbar grid item component
+   */
+  const ToolbarGridItem = styled(Grid)({
+    padding: "10px"
+  });
+
+  /**
+   * Toolbar grid container component
+   */
+  const ToolbarGridContainer = styled(Grid)({
+    alignContent: "space-around",
+    alignItems: "center"
+  });
+
+  /**
+   * Component properties
+   */
+  interface ToolbarTitleProps {
+    toolbarFormMode: ToolbarFormModes;
+  }
+
+  /**
+   * Toolbar title component
+   *
+   * @param props component properties
+   */
+  const ToolbarTitle = ({ toolbarFormMode }: ToolbarTitleProps) => {
+    const [title, setTitle] = useState(strings.tableToolbar.myRequests);
+
+    useEffect(() => {
+      if (toolbarFormMode) {
+        setTitle(getToolbarTitle(toolbarFormMode));
+      }
+    }, [toolbarFormMode]);
+    return <Typography variant="h5">{title}</Typography>;
+  };
```

Comment:
This should be in its own file or incorporated into the tableToolbar component. We should not have a component nested in another component like this

Reply:
Incorporated the toolbar title in the table toolbar